File: 4.8-bc/py/stage-2-feature-first/app/features/billing/repository.py
from __future__ import annotations

from sqlalchemy import text
from sqlalchemy.ext.asyncio import AsyncSession

from app.features.billing.model import Subscription


class Repository:
    def __init__(self, session: AsyncSession) -> None:
        self._session = session

    async def create(self, s: Subscription) -> None:
        await self._session.execute(
            text(
                "INSERT INTO billing_subscriptions "
                "(id, user_id, plan, next_charge_at, created_at) "
                "VALUES (:id, :user_id, :plan, :next_charge_at, :created_at)"
            ),
            {
                "id": s.id,
                "user_id": s.user_id,
                "plan": s.plan,
                "next_charge_at": s.next_charge_at,
                "created_at": s.created_at,
            },
        )
        await self._session.commit()

File: 4.8-bc/ts/stage-2-feature-first/src/features/billing/repository.ts
import type { DbPool } from "../../shared/db.js";
import type { Subscription } from "./model.js";

export class Repository {
  constructor(private readonly db: DbPool) {}

  async create(s: Subscription): Promise<void> {
    await this.db.query(
      `INSERT INTO billing_subscriptions (id, user_id, plan, next_charge_at, created_at)
       VALUES ($1, $2, $3, $4, $5)`,
      [s.id, s.userId, s.plan, s.nextChargeAt, s.createdAt],
    );
  }
}
